clients/vault/src/system.rs

```diff
@@ -293,19 +289,24 @@ where
 }
 
 impl VaultService {
-	fn new(
+	async fn new(
 		spacewalk_parachain: SpacewalkParachain,
 		config: VaultServiceConfig,
 		shutdown: ShutdownSender,
 	) -> Result<Self, Error> {
-		let stellar_wallet = StellarWallet::from_secret_encoded(&config.stellar_vault_secret_key)?;
+		let is_public_network = spacewalk_parachain.is_public_network().await?;
+
+		let mut stellar_wallet =
+			StellarWallet::from_secret_encoded(&config.stellar_vault_secret_key)?;
+		stellar_wallet.set_is_public_network(is_public_network);
 		let stellar_wallet = Arc::new(stellar_wallet);
+
```

Comment:
I assume you went the extra step of making the service creation async to ensure the wallet's `is_public_network` parameter is set correctly the moment it is created? I would then actually propose removing the `set_is_public_network` method altogether, and pass it into the the `from_secret_encoded` function. 

The reason why I added the `set_is_public_network`  in the first place is that I did not want to make the service creation async like you did but needed to set the network afterward. But generally we should not need to set the network after wallet creation so it's safe to remove.